crypto/modes/asm/aes-gcm-avx512.pl

```diff
@@ -0,0 +1,5141 @@
+  my $ZT17       = $_[25];    # [clobbered] ZMM register
+
+  # ; load current hash
+  $code .= "vmovdqu64         $CTX_OFFSET_AadHash($GCM128_CTX),$AAD_HASH\n";
+
+  &CALC_AAD_HASH($A_IN, $A_LEN, $AAD_HASH, $GCM128_CTX, $ZT0, $ZT1, $ZT2,
+    $ZT3, $ZT4, $ZT5, $ZT6, $ZT7, $ZT8, $ZT9, $ZT10, $ZT11, $ZT12, $ZT13,
+    $ZT14, $ZT15, $ZT16, $ZT17, $GPR1, $GPR2, $GPR3, $MASKREG);
+
+  # ; load current hash
+  $code .= "vmovdqu64         $AAD_HASH,$CTX_OFFSET_AadHash($GCM128_CTX)\n";
+}
+
+# ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+# ;;; Cipher and ghash of payloads shorter than 256 bytes
+# ;;; - number of blocks in the message comes as argument
+# ;;; - depending on the number of blocks an optimized variant of
+# ;;;   INITIAL_BLOCKS_PARTIAL is invoked
+sub GCM_ENC_DEC_SMALL {
+  my $AES_KEYS       = $_[0];     # [in] key pointer
+  my $GCM128_CTX     = $_[1];     # [in] context pointer
+  my $CIPH_PLAIN_OUT = $_[2];     # [in] output buffer
+  my $PLAIN_CIPH_IN  = $_[3];     # [in] input buffer
+  my $PLAIN_CIPH_LEN = $_[4];     # [in] buffer length
+  my $ENC_DEC        = $_[5];     # [in] cipher direction
+  my $DATA_OFFSET    = $_[6];     # [in] data offset
+  my $LENGTH         = $_[7];     # [in] data length
+  my $NUM_BLOCKS     = $_[8];     # [in] number of blocks to process 1 to 16
+  my $CTR            = $_[9];     # [in/out] XMM counter block
+  my $HASH_IN_OUT    = $_[10];    # [in/out] XMM GHASH value
+  my $INSTANCE_TYPE  = $_[11];    # [in] single or multi call
+  my $ZTMP0          = $_[12];    # [clobbered] ZMM register
+  my $ZTMP1          = $_[13];    # [clobbered] ZMM register
+  my $ZTMP2          = $_[14];    # [clobbered] ZMM register
+  my $ZTMP3          = $_[15];    # [clobbered] ZMM register
+  my $ZTMP4          = $_[16];    # [clobbered] ZMM register
+  my $ZTMP5          = $_[17];    # [clobbered] ZMM register
+  my $ZTMP6          = $_[18];    # [clobbered] ZMM register
+  my $ZTMP7          = $_[19];    # [clobbered] ZMM register
+  my $ZTMP8          = $_[20];    # [clobbered] ZMM register
+  my $ZTMP9          = $_[21];    # [clobbered] ZMM register
+  my $ZTMP10         = $_[22];    # [clobbered] ZMM register
+  my $ZTMP11         = $_[23];    # [clobbered] ZMM register
+  my $ZTMP12         = $_[24];    # [clobbered] ZMM register
+  my $ZTMP13         = $_[25];    # [clobbered] ZMM register
+  my $ZTMP14         = $_[26];    # [clobbered] ZMM register
+  my $ZTMP15         = $_[27];    # [clobbered] ZMM register
+  my $ZTMP16         = $_[28];    # [clobbered] ZMM register
+  my $ZTMP17         = $_[29];    # [clobbered] ZMM register
+  my $ZTMP18         = $_[30];    # [clobbered] ZMM register
+  my $ZTMP19         = $_[31];    # [clobbered] ZMM register
+  my $ZTMP20         = $_[32];    # [clobbered] ZMM register
+  my $ZTMP21         = $_[33];    # [clobbered] ZMM register
+  my $ZTMP22         = $_[34];    # [clobbered] ZMM register
```

Comment:
looks like $ZTMP15 to $ZTMP22 are unused in the subroutine, remove?

Reply:
Agree, removed.